pkg/agent/util/iptables/builder_test.go

```diff
@@ -0,0 +1,124 @@
+//go:build !windows
+// +build !windows
+
+// Copyright 2023 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iptables
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/util/intstr"
+
+	"antrea.io/antrea/pkg/apis/controlplane/v1beta2"
+)
+
+var (
+	ipsetAlfa  = "alfa"
+	ipsetBravo = "bravo"
+	eth0       = "eth0"
+	eth1       = "eth1"
+	port8080   = &intstr.IntOrString{Type: intstr.Int, IntVal: 8080}
+	port137    = &intstr.IntOrString{Type: intstr.Int, IntVal: 137}
+	port139    = int32(139)
+	port40000  = int32(40000)
+	port50000  = int32(50000)
+	icmpType0  = int32(0)
+	icmpCode0  = int32(0)
+)
+
+func TestBuilders(t *testing.T) {
+	testCases := []struct {
+		name      string
+		chain     string
+		buildFunc func(IPTablesRuleBuilder) IPTablesRuleBuilder
+		expected  string
+	}{
+		{
+			name:  "Accept TCP destination 8080 in FORWARD",
+			chain: ForwardChain,
+			buildFunc: func(builder IPTablesRuleBuilder) IPTablesRuleBuilder {
+				return builder.
+					MatchIPSetSrc(ipsetAlfa).
+					MatchIPSetDst(ipsetBravo).
+					MatchInputInterface(eth0).
+					MatchTransProtocol(v1beta2.ProtocolTCP).
+					MatchDstPort(port8080, nil).
+					SetComment(`"Accept TCP 8080"`).
```

Comment:
Are these double quotes required for the parameter? if yes, can we move it to inside of `SetComment`?

Reply:
Good idea!